File: runtime/go/units/wasi/args_sizes_get.go
// requires: memory/i32_store
func (w *WASI) wasi_args_sizes_get(argcPtr, bufSizePtr uint32) uint32 {
    total := uint32(0)
    for _, a := range w.args {
        total += uint32(len(a)) + 1
    }
    w.memory.i32_store(uint64(argcPtr), uint32(len(w.args)))
    w.memory.i32_store(uint64(bufSizePtr), total)
    return wasiOk
}

File: runtime/java/units/wasi/args_sizes_get.java
// requires: memory/i32_store
int wasi_args_sizes_get(int argcPtr, int bufSizePtr) {
    int total = 0;
    for (byte[] a : args) {
        total += a.length + 1;
    }
    memory.i32_store(Integer.toUnsignedLong(argcPtr), args.length);
    memory.i32_store(Integer.toUnsignedLong(bufSizePtr), total);
    return WASI_OK;
}
